llvm-external-projects/iree-dialects/lib/Dialect/IREEPyDM/Utils/TypeInference.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-dialects/Dialect/IREEPyDM/Utils/TypeInference.h"
+
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/TypeRange.h"
+
+using namespace mlir;
+namespace PYDM = mlir::iree_compiler::IREE::PYDM;
+using namespace PYDM;
+
+PermutedTypePropagator::PermutedBlockInfo *
+PermutedTypePropagator::addPermutedBlockToParent(ParentBlockInfo *parentInfo,
+                                                 Block *block) {
+  auto *permutedInfo = allocator.Allocate<PermutedBlockInfo>();
+  new (permutedInfo) PermutedBlockInfo();
```

Comment:
I think this can be written as:
`auto *permutedInfo = new (allocator) PermutedBlockInfo();`

Reply:
Thanks - that's what I was reaching for!